### PR TITLE
Add announcement render tests

### PR DESCRIPTION
**Pull request by PSwiecicki**

### CODE REVIEW

**Review by BartoszDombrow — APPROVED** [review]
Brak uwag ;)

File: frontend/src/components/Announcement/Announcement.test.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+describe('Announcement', () => {
+  const fakeAnnouncement = {
+    id: 0,
+    title: 'Skradziono rower!',
+    boardName: 'Sobótka',
+    content:
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris in risus eros. Morbi euismod magna ut lacus luctus tempus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Fusce leo eros, pretium et blandit vestibulum, accumsan nec tellus. Pellentesque odio est, posuere at lectus ac, fermentum dignissim purus. Pellentesque convallis faucibus ex sed sollicitudin. Vestibulum sodales quis felis eget facilisis. Integer venenatis, tellus in consectetur eleifend, risus nulla feugiat nisl, a ornare diam urna sed risus. Maecenas a volutpat est. Maecenas mattis odio a ligula euismod, sit amet viverra massa aliquet.\n',
+    comments: ['a', 'b', 'c'],
+    date: new Date(1995, 11, 17, 3, 24, 0),
```
Comment (PiotrRynio) [inline frontend/src/components/Announcement/Announcement.test.jsx:13]:
```suggestion
    date: new Date(),
```

File: frontend/src/components/Announcement/Announcement.test.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      </Router>,
+    );
+    const title = screen.getByText(fakeAnnouncement.title);
+    const boardName = screen.queryByText(fakeAnnouncement.boardName);
+    const commentsText = screen.getByText(`[${fakeAnnouncement.comments.length} comments]`);
+
+    // then
+    expect(title).toBeInTheDocument();
+    expect(boardName).not.toBeInTheDocument();
+    expect(commentsText).toBeInTheDocument();
```
Comment (PiotrRynio) [inline frontend/src/components/Announcement/Announcement.test.jsx:32]:
```suggestion

    // then
    const title = screen.getByText(fakeAnnouncement.title);
    const boardName = screen.queryByText(fakeAnnouncement.boardName);
    const commentsText = screen.getByText(`[${fakeAnnouncement.comments.length} comments]`);
    expect(title).toBeInTheDocument();
    expect(boardName).not.toBeInTheDocument();
    expect(commentsText).toBeInTheDocument();
```

File: frontend/src/components/Announcement/Announcement.test.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  });
+
+  it('Should properly render announcement with board name', async () => {
+    //given
+
```
Comment (PiotrRynio) [inline frontend/src/components/Announcement/Announcement.test.jsx:37]:
```suggestion
```

File: frontend/src/components/Announcement/Announcement.test.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    boardName: 'Sobótka',
+    content:
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris in risus eros. Morbi euismod magna ut lacus luctus tempus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Fusce leo eros, pretium et blandit vestibulum, accumsan nec tellus. Pellentesque odio est, posuere at lectus ac, fermentum dignissim purus. Pellentesque convallis faucibus ex sed sollicitudin. Vestibulum sodales quis felis eget facilisis. Integer venenatis, tellus in consectetur eleifend, risus nulla feugiat nisl, a ornare diam urna sed risus. Maecenas a volutpat est. Maecenas mattis odio a ligula euismod, sit amet viverra massa aliquet.\n',
+    comments: ['a', 'b', 'c'],
+    date: new Date(1995, 11, 17, 3, 24, 0),
+  };
+
+  it('Should properly render announcement', async () => {
+    //given
+
```
Comment (PiotrRynio) [inline frontend/src/components/Announcement/Announcement.test.jsx:18]:
```suggestion
```

File: frontend/src/components/Announcement/Announcement.test.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    //when
+    render(
+      <Router>
+        <Announcement announcement={fakeAnnouncement} showBoard={true} />
+      </Router>,
+    );
+    const boardName = screen.queryByText(fakeAnnouncement.boardName);
+
+    // then
+    expect(boardName).toBeInTheDocument();
```
Comment (PiotrRynio) [inline frontend/src/components/Announcement/Announcement.test.jsx:47]:
```suggestion

    // then
    const boardName = screen.queryByText(fakeAnnouncement.boardName);
    expect(boardName).toBeInTheDocument();
```

File: frontend/src/components/Announcement/Announcement.test.jsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      <Router>
+        <Announcement announcement={fakeAnnouncement} isOpen={true} />
+      </Router>,
+    );
+    const commentsText = screen.queryByText(`[${fakeAnnouncement.comments.length} comments]`);
+    const content = screen.getByText(/Vestibulum sodales quis felis eget facilisis/i);
+
+    // then
+    expect(content).toBeInTheDocument();
+    expect(commentsText).not.toBeInTheDocument();
```
Comment (PiotrRynio) [inline frontend/src/components/Announcement/Announcement.test.jsx:64]:
```suggestion

    // then
    const commentsText = screen.queryByText(`[${fakeAnnouncement.comments.length} comments]`);
    const content = screen.getByText(/Vestibulum sodales quis felis eget facilisis/i);
    expect(content).toBeInTheDocument();
    expect(commentsText).not.toBeInTheDocument();
```

**Review by PiotrRynio — APPROVED** [review]

File: frontend/src/components/Announcement/Announcement.test.jsx (hunk 1 above)
Comment (PSwiecicki) [reply frontend/src/components/Announcement/Announcement.test.jsx:13]: doone

File: frontend/src/components/Announcement/Announcement.test.jsx (hunk 3 above)
Comment (PSwiecicki) [reply frontend/src/components/Announcement/Announcement.test.jsx:37]: done

File: frontend/src/components/Announcement/Announcement.test.jsx (hunk 4 above)
Comment (PSwiecicki) [reply frontend/src/components/Announcement/Announcement.test.jsx:18]: done

File: frontend/src/components/Announcement/Announcement.test.jsx (hunk 5 above)
Comment (PSwiecicki) [reply frontend/src/components/Announcement/Announcement.test.jsx:47]: done

**Pull request merged**